lib/Doctrine/DBAL/Query/QueryBuilder.php

```diff
@@ -201,14 +201,16 @@ public function getState()
     /**
      * Executes this query using the bound parameters and their types.
      *
-     * @return ResultStatement|int
+     * @return ForwardCompatibility\DriverStatement|int
      *
      * @throws Exception
      */
     public function execute()
     {
         if ($this->type === self::SELECT) {
-            return $this->connection->executeQuery($this->getSQL(), $this->params, $this->paramTypes);
+            return ForwardCompatibility\Result::ensure(
+                $this->connection->executeQuery($this->getSQL(), $this->params, $this->paramTypes)
+            );
```

Comment:
Ensuring compatible result here as 2 advantages :
- We can be more accurate on return type documentation, otherwise we will have to document ForwardCompatibility\DriverStatement|ForwardCompatibility\DriverResultStatement|int
- If user has overridden Connection::executeQuery() the forward compatibility will be ensured